common/src/main/java/com/wynntils/features/redirects/ChatRedirectFeature.java

```diff
@@ -503,10 +505,13 @@ protected StyledText getNotification(Matcher matcher) {
     }
 
     private class LoginRedirector extends SimpleRedirector {
+        // Test suite: https://regexr.com/7jhti
+        private static final String RANK_STRING =
+                Arrays.stream(PlayerRank.values()).map(PlayerRank::getTag).collect(Collectors.joining());
         private static final Pattern FOREGROUND_PATTERN = Pattern.compile(
-                "^§.\\[(§.)+\\|?(§.)*(?<rank>[a-zA-Z+]+)(§.)+\\|?(§.)*\\] §.(?<name>.*)§. has just logged in!$");
-        private static final Pattern BACKGROUND_PATTERN = Pattern.compile(
-                "^(?:§8)?\\[(§.)+\\|?(§.)*(?<rank>[a-zA-Z+]+)(§.)+\\|?(§.)*\\] §7(?<name>.*)§8 has just logged in!$");
+                "^(?<rank>[" + RANK_STRING + "]) §#[0-9a-f]{5}(?<name>[\\w ]{1,20})§[0-9a-f] has just logged in!$");
+        private static final Pattern BACKGROUND_PATTERN = Pattern.compile("^(?:§8)?\\[(§.)+\\|?(§.)*(?<rank>["
+                + RANK_STRING + "]) §#[0-9a-f]{5}(?<name>[\\\\w ]{1,20})§[0-9a-f] has just logged in!$");
```

Comment:
`§#[0-9a-f]{5}`?

Reply:
I double don't understand this? 5 chars? Not 6/8? I know we use `§#` in StyledText to represent colors outside of the ChatFormatting range. But I am not sure what this is..

Reply:
Yeah I dunno either. This is what I get from chat.
`[03:40:50] [Render thread/INFO]: [CHAT]  &#fe600v8j&6 has just logged in!`